src/main/java/cz/fi/muni/pa165/Model/Entity/ApplicationApprovedRecord.java

```diff
@@ -59,4 +60,30 @@ public Date getTo() {
     public RentApplication getApplication() {
         return application;
     } 
+    
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if ((o == null) || !(o instanceof ApplicationApprovedRecord)) {
+            return false;
+        }
+        ApplicationApprovedRecord other = (ApplicationApprovedRecord) o;
+        if (getFrom() != null ? !getFrom().equals(other.getFrom()) : other.getFrom() != null) {
+            return false;
+        }
+        if (getTo() != null ? !getTo().equals(other.getTo()) : other.getTo() != null) {
+            return false;
+        }
+        if (application != null ? !application.equals(other.getApplication()) : other.getApplication() != null) {
+            return false;
+        }
+	return true;
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder().append(getFrom()).append(getTo()).append(getApplication().getId()).toHashCode();
```

Comment:
We got into a funny situation here. As you mentioned, this is a team project, so @jkuchar doesn't decide for all of us, we have to come to a consensus together, or at least try to vote for an approach we want in our code. I guess we all can agree we want to have a consistent code. This means we should take one approach and implement it equally in all entities.  
I think this can be solved by a simple vote (I can be wrong, stop me in that case).  
I personally stand in this with @jkuchar , it seems to me that ID is enough for us to identify entities with. And I am not changing my point of view without further arguments for why use business keys indead.